src/main/java/com/uid2/shared/store/reader/RotatingKeyAclProvider.java

```diff
@@ -18,7 +20,32 @@ public class RotatingKeyAclProvider implements IKeyAclProvider, StoreReader<Map<
     private final ScopedStoreReader<AclSnapshot> reader;
 
     public RotatingKeyAclProvider(DownloadCloudStorage fileStreamProvider, StoreScope scope) {
-        this.reader = new ScopedStoreReader<>(fileStreamProvider, scope, new KeyAclParser(), "key acls");
```

Comment:
This will cause issues with things that expect a normal reader. We should leave this constructor unchanged and then just create the encrypted reader in the other constructor.